### ISSUE TITLE
Detach instance when spot termination notification is received  - event is not from CloudWatch but from SNS

### ISSUE THREAD

**Issue by mello7tre** [AutoSpotting/AutoSpotting#322]
# Github issue #

## Issue type ##

- Bug Report

## Build number #

master

## Environment ##

- AWS region
- Type of environment: (VPC, EC2Classic or DefaultVPC)
- Anonymized launch configuration: 

```text
```

## Summary ##

In core/spot_termination.go func GetInstanceIDDueForTermination (line 43).

Event is not coming from CloudWatch but from SNS!

Code must be changed to extract from the SNS message the CloudWatch event json, before parsing it.
Currently when a spot instance terminate Autospotting lambda is executed but give error:
```
           START RequestId: 087eba8f-99e0-4383-ade8-69f33b7b3ed7 Version: $LATEST
           2019/02/13 07:32:29 unexpected end of JSON input
           END RequestId: 087eba8f-99e0-4383-ade8-69f33b7b3ed7
```

## Expected results ##

Lambda is run and instance detached.

## Actual results ##

An error is displayed in log.

**Comment by mello7tre** [comment]
Changes to do are deeper:
must be changed Handler func in autospotting.go too

and must be found a way to extract region from sns EventSource or TopicArn.

**Comment by cristim** [comment]
Thanks for reporting this.

I am aware of this problem, the detachment logic is relatively new and needs to mature a bit.

Please feel free to create a PR with a fix and I would gladly consider it.

**Comment by mello7tre** [comment]
I have fixed it and tested it publishing directly to sns, with a message like then one sent by the  cloudwatch eventrule.

I will leave it running in my env until an instance is terminated to fully test it.
Then i will open a PR.

**Comment by cristim** [comment]
Fixed on #323
